docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+```
+
+The code above defines a singleton type called `stats_singleton`. The singleton, is a special table that stores statistical data which are encapsulated in the `statsdata` structure. The structure contains just one field `hi_count` which counts the number of time the `hi` action was executed.
+
+Developers can use the `stats_singleton` template type, to instantiate a reference to the singleton table and to perform various operations, such as:
+
+- read the singleton data,
+- modify existing singleton data,
+- delete existing singleton data.
+
+### Singleton: Code, Scope and Table
+
+The code, scope and table have the same same meaning as those for the [multi-index table](#multi-index-code-scope-and-table).
+
+#### Get Singleton Data
+
+This is how you get the singleton data:
+
+```cpp
+stats_singleton stats(get_self(), get_self().value);
+
+if (stats.exists()) {
+    auto current_stats = stats.get();
+    print_f("Stats for `hi` action: %\n", current_stats.hi_count);
+}
+```
+
+#### Modify Singleton Data
+
+This is how you modify the singleton data:
+
+```cpp
+stats_singleton stats(get_self(), get_self().value);
+
+auto current_stats = stats.get_or_create(get_self(), {0});
+current_stats.hi_count += 1;
+stats.set(current_stats, get_self());
+```
+
+#### Delete Singleton Data
+
+This is how you delete the singleton data:
+
+```cpp
+stats_singleton stats(get_self(), get_self().value);
+
+if (stats.exists()) {
+    stats.remove();
+    print_f("Stats have been removed.");
+}
+```
+
+### Extend Hello Smart Contract with Singleton
+
+Now, that you know the basic operations you can perform on a singleton, you can extend the `hello` contract `hi` action to:
+
+- Implement a `stats` singleton which stores statistical data about the `hi` action number of executions,
+- Implement `getstats` action to read the stats singleton,
+- Implement `deletestats` action to delete the stats singleton.
+
+The hello.hpp file should look like this:
+
+```cpp
+#include <eosio/eosio.hpp>
+#include <eosio/asset.hpp>
+#include <eosio/singleton.hpp>
+
+using namespace eosio;
+
+CONTRACT hello : public contract {
+   public:
+      using contract::contract;
+
+      ACTION hi(name nm);
+      ACTION gettime();
+      ACTION goodbye(name nm);
+      ACTION getstats();
+      ACTION deletestats();
+
+   private:
+      void parse_time(uint64_t epoch_time, uint16_t& year, uint8_t& month, uint8_t& day, uint8_t& hour, uint8_t& minute, uint8_t& second);
+
+      TABLE user_balance {
+         name user;
+         asset balance;
+
+         uint64_t primary_key() const { return user.value; }
+      };
+      using balance_table = eosio::multi_index<"balances"_n, user_balance>;
+
+      TABLE statsdata {
+         int hi_count;
+      };
+      using stats_singleton = eosio::singleton<"stats"_n, statsdata>;
+};
+```
+
+The `hi`, `getstats` and `deletestats` actions implementation should look like this:
+
+```cpp
+ACTION hello::hi(name nm) {
```

Comment:
Don't need to post the ENTIRE contract again each time, just the additions. It'll make the guide much shorter. 

The guide's scrollbar handle could intimidate mike tyson.

Reply:
fixed